pkg/tagset/null_factory_test.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.Datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// NOTE: this file is named *_test.go because it is only intended for use in
+// tests within this package.
+
+package tagset
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/twmb/murmur3"
+)
+
+func TestNullFactory(t *testing.T) {
+	testFactory(t, func() Factory { return newNullFactory() })
+}
+
+// A nullFactory caches nothing. It is useful for tests that need a factory.
```

Comment:
Why tests would not be able to use a caching factory instead?

Reply:
This helps to isolate behavior in tests (so that cached results aren't used by mistake, for example).

Reply:
Would it be useful to have a reset on the caching factory internal state and avoid the duplicated code here?